src/Components/OpenInExploreLogsButton/OpenInExploreLogsButton.tsx

```diff
@@ -0,0 +1,76 @@
+import { config, useReturnToPrevious } from '@grafana/runtime';
+import { LinkButton } from '@grafana/ui';
+import React, { useMemo } from 'react';
+import { escapeURLDelimiters, replaceEscapeChars, replaceSlash, stringifyAdHocValues } from 'services/extensions/links';
+import { OpenInExploreLogsButtonProps } from './types';
+import { AbstractLabelOperator } from '@grafana/data';
+import { LabelFilterOp } from 'services/filterTypes';
+
+const operatorMap = {
+  [AbstractLabelOperator.Equal]: LabelFilterOp.Equal,
+  [AbstractLabelOperator.NotEqual]: LabelFilterOp.NotEqual,
+  [AbstractLabelOperator.EqualRegEx]: LabelFilterOp.RegexEqual,
+  [AbstractLabelOperator.NotEqualRegEx]: LabelFilterOp.RegexNotEqual,
+};
+
+export default function OpenInExploreLogsButton({
+  datasourceUid,
+  labelMatchers,
+  from,
+  to,
+  returnToPreviousSource,
+  renderButton,
+}: OpenInExploreLogsButtonProps) {
+  const setReturnToPrevious = useReturnToPrevious();
+
+  const href = useMemo(() => {
+    const mainLabel = labelMatchers[0];
+
+    if (
+      !mainLabel ||
+      // we can't open in explore logs if main label matcher is smth different from equal
+      mainLabel?.operator !== AbstractLabelOperator.Equal
+    ) {
+      return null;
+    }
+
+    const url = new URL(
+      `${config.appSubUrl || config.appUrl}a/grafana-lokiexplore-app/explore/${mainLabel.name}/${replaceSlash(
+        mainLabel.value
+      )}/logs`
+    );
+
+    datasourceUid && url.searchParams.set('var-datasource', datasourceUid);
+    from && url.searchParams.set('from', from);
+    to && url.searchParams.set('to', to);
+
+    labelMatchers.forEach((labelMatcher) => {
+      url.searchParams.append(
+        'var-filters',
```

Comment:
Also nit, but for `filters` we could import `VAR_LABELS` from `services/variables`. That would make it a bit more resilient.